eng/Subsets.props

```diff
@@ -119,7 +119,9 @@
 
   <PropertyGroup>
     <!-- CLR NativeAot only builds in a subset of the matrix -->
-    <NativeAotSupported Condition="('$(TargetOS)' == 'windows' or '$(TargetOS)' == 'linux' or '$(TargetOS)' == 'osx' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'iossimulator' or '$(TargetOS)' == 'ios' or '$(TargetOS)' == 'tvossimulator' or '$(TargetOS)' == 'tvos' or '$(TargetOS)' == 'freebsd') and ('$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm')">true</NativeAotSupported>
+    <_NativeAotSupportedOS Condition="'$(TargetOS)' == 'windows' or '$(TargetOS)' == 'linux' or '$(TargetOS)' == 'osx' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'iossimulator' or '$(TargetOS)' == 'ios' or '$(TargetOS)' == 'tvossimulator' or '$(TargetOS)' == 'tvos' or '$(TargetOS)' == 'freebsd'">true</_NativeAotSupportedOS>
```

Comment:
Looks like we can invert it `NativeAotNotSupported`; that set seems to be getting smaller. 😅

Reply:
Seems like the unsupported OSes are `tizen`, `wasi`, and `browser`. Not sure of the status for `haiku`, `illumos`, `solaris` and `netbsd` which happen to have some build system support.

Unsupported archs are RV64, LA64, PowerPC (supported by Mono) and linux-x86. We may be able to shrink this list before .NET 9 ships.

Reply:
Yup, something like:
```csproj
<_NativeAotUnsupportedOS>browser,haiku,illumos,netbsd,tizen,wasi</_NativeAotUnsupportedOS>
<NativeAotUnsupportedOS Condition="$(_NativeAotUnsupportedOS.Contains('$(TargetOS)'))">false</NativeAotUnsupportedOS>
```

usage: `Condition="'$(NativeAotUnsupportedOS)' != 'false'"`

> Unsupported archs are RV64, LA64, PowerPC (supported by Mono), WASM, and linux-x86

and s390x.

Reply:
FWIW, this only needs to create a subset of what _can already build_ with CoreCLR but _does not build_ for native AOT (so that we don't break `./build.sh clr` for a platform that already works otherwise).

I don't think browser/wasi/powerpc/s390 _builds_ with CoreCLR, so we don't mind if native AOT makes `./build.sh clr` more broken. If someone brings up a new platform and wants native AOT out of the way, they can exclude it, same way they can exclude anything else they don't (yet) want as part of bringup.

Reply:
> I don't think browser/wasi/powerpc/s390 builds with CoreCLR,

Given we have to ensure that clr.iltools and clr.packages remain intact:
https://github.com/dotnet/runtime/blob/3eb8c7f1086b79b28a27b57a935f97be3b7fcccb/eng/pipelines/runtime-community.yml#L66

we might as well just list the 'not yet supported' ones without classifying them further.

Reply:
> Given we have to ensure that clr.iltools and clr.packages remain intact:

Clr.iltools "just works" because clr.iltools doesn't include coreclr VM and doesn't include any native AOT parts either, same for clr.packages.

Does building the clr subset (the entire clr subset, not just ilasm/dasm) work on s390 today? If not, I don't see why we'd need an extra exclusion for the native aot part. We shouldn't need to add extra exclusions (that are _specific to native AOT_) every time someone adds an obscure platform for mono. We either skip the entire clr build (i.e. we do the thing where build is skipped for all of coreclr, not just native AOT - except for iltools), or let it do whatever it does (break, probably). I think the latter is fine because that's how it is today.